added uzp as Universal Zip Package, improved regexp (#7)
Gibberlings3/modhub: README.md

# modhub
Microservice to find the latest release, prerelease, or latest commit of a mod hosted on GitHub.

It will automatically redirect to the desired archive. Defaults to archived code of the latest proper release.

# Usage

> Construct the desired url and open it inside browser or use for commandline tool.

Reviewer: ```suggestion
Construct the desired URL and open it with a browser or other tool.
```

Spell Revisions latest prerelease:  
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&preonly

Spell Revisions latest release or prerelease, whichever of them is newer:  
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&pre

Spell Revisions latest master commit (unpackaged):  
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&master

Spell Revisions latest release or prerelease, if there is no release or prerelease then fallback to the last commit from master branch:  
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&ifeellucky

Spell Revisions universal zip package from latest release:
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&pkg=uzp

Spell Revisions windows zip package from latest release:
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&pkg=win

Spell Revisions macOS zip package from latest prerelease:
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&pkg=osx&preonly

Spell Revisions Linux zip package from latest release or prerelease, whichever of them is newer:
http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&pkg=lin&pre

# Usage example

> Get the link of the most recent macOS package for SpellRevisions release pre-release, whichever of them is newer:

Reviewer: ```suggestion
Get the link to the most recent macOS (pre-)release package of SpellRevisions:
```

    curl -I "https://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&pkg=osx&pre"
    HTTP/1.1 302 Moved Temporarily
    Date: Tue, 25 Jun 2019 06:48:56 GMT
    Server: Apache
    Upgrade: h2,h2c
    Connection: Upgrade
    Location: https://github.com/Gibberlings3/SpellRevisions/releases/download/v4b16/osx-spell_rev-v4-beta16.zip
    Vary: User-Agent
    Content-Type: text/html; charset=UTF-8

The URL printed under Location is the file that you want.

# For mod authors
If you want the package detection to work, make sure you attach packages to the Github release and mind these patterns:
 * universal zip package name should end with ".zip"  
 WARNING: if there are multiple .zip files in the release (excluding those with 'win-', 'osx-', 'lin-' prefixes), the behaviour is undefined.
 * windows exe package name should end with ".exe"
 * windows zip package name should start with "win-" and end with ".zip"
 * macOS package name should start with "osx-"
 * linux package name should start with "lin-"

> For an real life example, check the [release](https://github.com/Gibberlings3/Tweaks-Anthology/releases/latest) of Tweaks Anthology.

Reviewer: ```suggestion
For a real life example, check a [release](https://github.com/Gibberlings3/Tweaks-Anthology/releases/latest) of Tweaks Anthology.
```

http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions&ifeellucky

Spell Revisions universal zip package from latest release:

> http://lynxlynx.info/ie/modhub.php?InfinityMods/ModPackage&pkg=uzp

Reviewer: This returns an html doc for me, not a zip. Plus the explanation of the url still mentions spell revisions.

Author: Fixed.
The 'win' works: https://lynxlynx.info/ie/modhub.php?InfinityMods/ModPackage&pkg=win but the uzp code is not yet committed so it doesn't work yet.

# For mod authors
If you want the package detection to work, make sure you attach packages to the Github release and mind these patterns:
 * a universal zip package name should end with "zip"
 * a windows exe package name should end with "exe"
 * a windows zip package name should start with "win" and end with "zip"
 * a linux package name should start with "lin"
 * an osx package name should start with "osx"
 

> WARNING: for uzip, if there are multiple zip files (except those with 'win-', 'osx-', 'lin-' prefixes) in the release, behaviour is undefined.

Reviewer: ```suggestion
WARNING: for uzp, if there are multiple zip files (except those with 'win-', 'osx-', 'lin-' prefixes) in the release, behaviour is undefined.
```

> WARNING: for uzp, if there are multiple .zip files (except those with 'win-', 'osx-', 'lin-' prefixes) in the release, behaviour is undefined.

Reviewer: This warning would better fit in the main section above, since it's more for users than modders. It will also unbreak the flow of thought, since Tweaks Anthology isn't an example of this.

An example for the example section would be helpful too.

Author: Regarding warning - will do.
Does the example for uzp should based on `http://lynxlynx.info/ie/modhub.php?Gibberlings3/SpellRevisions` even if the mod doesn't have them or I should use working example?

Reviewer: A working one, of course, that's the point of an example — it can be tried successfully.

Author: done

 * macOS package name should start with "osx-"
 * linux package name should start with "lin-"

> For a real life example, check a [release](https://github.com/Gibberlings3/Tweaks-Anthology/releases/latest) of Tweaks Anthology.

Reviewer: come on, it was perfect and now you've introduced an error.

Author: done

> For an real life example, check [latest release](https://github.com/Gibberlings3/Tweaks-Anthology/releases/latest) of Tweaks Anthology.

Reviewer: please revert this, but there's also the stray "an".

Author: done but I'm not sure if this is what you wanted

Reviewer: revert = undo. The github release stream has all releases ever made, so just looking at the last may not show everything. I'll add a suggestion to the change.

Author: done
